Sources/CvPlot.cpp

```diff
@@ -1759,30 +1649,20 @@ bool CvPlot::shareAdjacentArea(const CvPlot* pPlot) const
 {
 	PROFILE_FUNC();
 
-	int iCurrArea;
-	int iLastArea;
-	CvPlot* pAdjacentPlot;
-	int iI;
+	int iLastArea = FFreeList::INVALID_INDEX;
 
-	iLastArea = FFreeList::INVALID_INDEX;
-
-	for (iI = 0; iI < NUM_DIRECTION_TYPES; ++iI)
+	foreach_(CvPlot* adjacentPlot, adjacent())
 	{
-		pAdjacentPlot = plotDirection(getX_INLINE(), getY_INLINE(), ((DirectionTypes)iI));
+		const int iCurrArea = adjacentPlot->getArea();
 
-		if (pAdjacentPlot != NULL)
+		if (iCurrArea != iLastArea)
 		{
-			iCurrArea = pAdjacentPlot->getArea();
-
-			if (iCurrArea != iLastArea)
+			if (pPlot->isAdjacentToArea(iCurrArea))
 			{
-				if (pPlot->isAdjacentToArea(iCurrArea))
-				{
-					return true;
-				}
-
-				iLastArea = iCurrArea;
+				return true;
 			}
+
+			int iLastArea = iCurrArea;
```

Comment:
Woah!! That is entirely broken!!!

Reply:
What it was doing?

Reply:
Meaning I broke it or it was broken before?

Reply:
Its very clear you broke it, just read the code!

Reply:
cause I added int again to iLastArea?

Reply:
Yeah you declared a new variable `iLastArea` and set it to `iCurrArea`.